stable/datadog/templates/daemonset.yaml

```diff
@@ -0,0 +1,56 @@
+{{- if .Values.datadog.apiKey }}
+apiVersion: extensions/v1beta1
+kind: DaemonSet
```

Comment:
@gtaylor  If you set the environment variable SD_BACKEND=docker it will enable our [service discovery functionality](https://www.datadoghq.com/blog/service-discovery-monitoring-docker-containers-move/) in the agent such that it automatically monitors new containers for the correct applications as they are started/stopped/rescheduled.

My comment on [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) relates to the kubernetes_state integration which was added in the 5.10 release. If a kube-state-metrics pod is online and seen by service discovery, we'll pull in all the metrics that kube-state-metrics exposes.